crates/ruff_linter/src/rules/pyflakes/rules/f_string_missing_placeholders.rs

```diff
@@ -47,11 +47,12 @@ impl AlwaysFixableViolation for FStringMissingPlaceholders {
 
 /// F541
 pub(crate) fn f_string_missing_placeholders(checker: &mut Checker, expr: &ast::ExprFString) {
-    if expr
-        .value
-        .f_strings()
-        .any(|f_string| f_string.values.iter().any(Expr::is_formatted_value_expr))
-    {
+    if expr.value.f_strings().any(|f_string| {
+        f_string
+            .elements
+            .iter()
+            .any(ast::FStringElement::is_expression)
+    }) {
```

Comment:
Nit: We could add these as a method on `fstring` itself: `has_placeholders()` or similar if we want.

Reply:
I thought so but currently it's only being used in a single place.